RecoEgamma/EgammaPhotonProducers/src/ReducedEGProducer.cc

```diff
@@ -311,65 +321,104 @@ void ReducedEGProducer::produce(edm::Event& theEvent, const edm::EventSetup& the
     if (relink) superClusterFullRelinkMap.insert(mappedscidx);
-    
+
     //conversions only for full relinking
     if (!relink) continue;
-    
+
     const reco::ConversionRefVector &convrefs = photon.conversions();
     for (const reco::ConversionRef &convref : convrefs) {
       if (!conversionMap.count(convref)) {
         conversions->push_back(*convref);
         conversionMap[convref] = conversions->size() - 1;
       }
     }
-    
+
     //explicitly references conversions
     const reco::ConversionRefVector &singleconvrefs = photon.conversionsOneLeg();
     for (const reco::ConversionRef &convref : singleconvrefs) {
       if (!singleConversionMap.count(convref)) {
         singleConversions->push_back(*convref);
         singleConversionMap[convref] = singleConversions->size() - 1;
       }
-    }    
-    
+    }
+
   }
-  
+
   //loop over electrons and fill maps
   for (unsigned int iele = 0; iele<gsfElectronHandle->size(); ++iele) {
     const reco::GsfElectron &gsfElectron = (*gsfElectronHandle)[iele];
-    
-    bool keep = keepGsfElectronSel_(gsfElectron);    
+
+    bool keep = keepGsfElectronSel_(gsfElectron);
     if (!keep) continue;
-    
+
     reco::GsfElectronRef gsfElectronref(gsfElectronHandle,iele);
-    
+
     gsfElectrons->push_back(gsfElectron);
     pfCandIsoPairVecEle.push_back((*gsfElectronPfCandMapHandle)[gsfElectronref]);
-    
+
     //fill electron id valuemap vectors
     for (unsigned int iid=0; iid<gsfElectronIds.size(); ++iid) {
       gsfElectronIdVals[iid].push_back( (*gsfElectronIdHandles[iid])[gsfElectronref] );
-    }    
+    }
 
     for (unsigned int iid=0; iid<gsfElectronPFClusterIsos.size(); ++iid) {
       gsfElectronPFClusterIsoVals[iid].push_back( (*gsfElectronPFClusterIsoHandles[iid])[gsfElectronref] );
-    }    
+    }
 
     const reco::GsfElectronCoreRef &gsfElectronCore = gsfElectron.core();
     if (!gsfElectronCoreMap.count(gsfElectronCore)) {
       gsfElectronCores->push_back(*gsfElectronCore);
       gsfElectronCoreMap[gsfElectronCore] = gsfElectronCores->size() - 1;
-    }    
-    
+    }
+
+
+    // ----------------
+    //gsftracks, add by me - Hien xinh dep
+
+    const reco::GsfTrackRef &gsfTrack = gsfElectron.gsfTrack();
+
+    // Save the main gsfTrack
+    if (!gsfTrackMap.count(gsfTrack)) {
+      gsfTracks->push_back(*gsfTrack);
+      gsfTrackMap[gsfTrack] = gsfTracks->size() - 1;
+      //std::cout << "push back gsftrack to gsftrack ref" << std::endl;
+      //std::cout << " gsf track size: " << gsfTracks->size() << std::endl;
+    }
+
+
+    //std::cout << " ---- event number: " << theEvent.id().event() << "\t gsf pt: " << gsfTrack->pt() << "\t gsf eta: "
+    //	      << gsfTrack->eta() << "\t gsf phi: " << gsfTrack->phi() << std::endl;
+    //std::cout << "ambiguous true or false: " <<gsfElectron.ambiguous()
+    //	      << "  (looks like this flag was never set even when there are ambiguous tracks present)" <<std::endl;
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant